### PR TITLE
New: Add `vue/name-property-required` rule

### PR DESCRIPTION
**Pull request by LukeeeeBennett**
Require that a `name` property is set.

Originally wanted to handle name value being an `Identifier` but I'm not sure its worth the hassle as I've only ever written `Literal` name values and I can't think of any critical case that would require an `Identifier` value type.  Please let me know if this is a bad assumption.

Also let me know if you'd prefer just to check that the property key exists and not that the value is also a truthy `Literal`, understandable.

### CODE REVIEW

**Comment by LukeeeeBennett** [comment]
I'm not sure why the netlify builds failed because there is no error message but non-zero exit code :man_shrugging:

**Comment by LukeeeeBennett** [comment]
*mic tap*

File: lib/rules/name-property-required.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
+/**
+ * @fileoverview Require a name property in Vue components
+ * @author LukeeeeBennett
+ */
+'use strict'
+
+const utils = require('../utils')
+
+function isNameProperty (node) {
+  return node.type === 'Property' && node.key.name === 'name'
```
Comment (ota-meshi) [inline lib/rules/name-property-required.js]: Please check if it is computed property.

File: tests/lib/rules/name-property-required.js [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+  valid: [
+    {
+      filename: 'ValidComponent.vue',
+      code: `
+        export default {
+          name: 'IssaName'
+        }
+      `,
+      parserOptions
+    }
```
Comment (ota-meshi) [inline tests/lib/rules/name-property-required.js]: Could you add the following testcase.

```js
export default {
    [name]: 'Name'
}
```

File: lib/rules/name-property-required.js [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+}
+
+function hasTruthyLiteralValue (node) {
+  return node.value.type === 'Literal' && node.value.value
```
Comment (ota-meshi) [inline lib/rules/name-property-required.js]: I think that another rule is necessary to verify whether it is Literal.
What do you think?

File: lib/rules/name-property-required.js (hunk 1 above)
Comment (LukeeeeBennett) [reply lib/rules/name-property-required.js]: It now ignores computed properties, is this what you wanted?

File: tests/lib/rules/name-property-required.js (hunk 2 above)
Comment (LukeeeeBennett) [reply tests/lib/rules/name-property-required.js]: Thanks :)

File: lib/rules/name-property-required.js (hunk 3 above)
Comment (LukeeeeBennett) [reply lib/rules/name-property-required.js]: Sounds like a good idea yes :) I will open a different PR for it and link here

**Comment by ota-meshi** [comment]
@LukeeeeBennett Hello!
`npm install` failed because the data cached in circleci is broken. It's not your fault.

**Pull request merged**